dspace-spring-rest/src/main/java/org/dspace/app/rest/exception/DSpaceApiExceptionControllerAdvice.java

```diff
@@ -80,21 +83,70 @@ protected void ParameterConversionException(HttpServletRequest request, HttpServ
 
     @Override
     protected ResponseEntity<Object> handleMissingServletRequestParameter(MissingServletRequestParameterException ex,
-            HttpHeaders headers, HttpStatus status, WebRequest request) {
+                                                                          HttpHeaders headers, HttpStatus status,
+                                                                          WebRequest request) {
         // we want the 422 status for missing parameter as it seems to be the common behavior for REST application, see
-        // https://stackoverflow.com/questions/3050518/what-http-status-response-code-should-i-use-if-the-request-is-missing-a-required
+        // https://stackoverflow.com/questions/3050518/what-http-status-response-code-should-i-use-if-the-request-is
+        // -missing-a-required
         return super.handleMissingServletRequestParameter(ex, headers, HttpStatus.UNPROCESSABLE_ENTITY, request);
     }
 
     @Override
     protected ResponseEntity<Object> handleTypeMismatch(TypeMismatchException ex, HttpHeaders headers,
-            HttpStatus status, WebRequest request) {
+                                                        HttpStatus status, WebRequest request) {
         // we want the 422 status for type mismatch on parameters as it seems to be the common behavior for REST
         // application, see
-        // https://stackoverflow.com/questions/3050518/what-http-status-response-code-should-i-use-if-the-request-is-missing-a-required
+        // https://stackoverflow.com/questions/3050518/what-http-status-response-code-should-i-use-if-the-request-is
+        // -missing-a-required
         return super.handleTypeMismatch(ex, headers, HttpStatus.UNPROCESSABLE_ENTITY, request);
     }
 
+    @ExceptionHandler(Exception.class)
```

Comment:
I think this is the root case of the test failures. You need to take into account that some exceptions have custom response codes defined on them. For example [PatchBadRequestException](https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/exception/PatchBadRequestException.java)

I would change the code in the method below to look for the `@ResponseStatus`  annotation (you can [find an example here](https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/model/hateoas/DSpaceRelProvider.java#L24)). If you find it, use the response value specified there (field `value`). If no annotation was found, use the `HttpServletResponse.SC_INTERNAL_SERVER_ERROR`. 

I think that will solve your tests.

Reply:
The issue was that if exceptions weren't caught in this manner (so if we just threw the error without ever catching it anywhere), there wouldn't be any CORS headers visible in the response and a few other things went missing as well. If they're being caught in this class and being handled by the sendErrors method that's always called, the response does contain all the wanted information.
This was the reasoning for adding the general Exception.class ExceptionHandler so that the expected headers are always present
This does indeed break a few tests and I've already tried to fix most of them by adding more and more specific ExceptionHandlers to catch all the specific errors that are thrown and then 're-throwing' them in the proper fashion with all the wanted response headers.

However if I'm understanding your suggestion correctly, you're suggesting to keep a general ExceptionHandler and then looking for an @ResponseStatus annotation for the Exception that's thrown/caught.. That would indeed solve all the test cases as the correct status would be returned in the response and it does simplify the entire class because we wouldn't need to declare an ExceptionHandler for every single custom thrown Exception. I'll look into this as soon as I have time, thank you for the suggestion!